From ffcdaf7f2a2ef712931e90e2f4dd27a7417ece59 Mon Sep 17 00:00:00 2001
From: Lei Qian <lei.qian@amlogic.com>
Date: Thu, 27 Sep 2018 12:38:53 +0800
Subject: [PATCH] FallbackHome: set tv_user_setup_complete to 1 when firstboot
 [1/1]

PD#173290

    only set it in AOSP version, because we have not user setup APK
in AOSP version

Change-Id: I009fe829bb0affab63f6833832dd60b55ad3fe79
Signed-off-by: Lei Qian <lei.qian@amlogic.com>
---
 Settings/src/com/android/tv/settings/system/FallbackHome.java | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Settings/src/com/android/tv/settings/system/FallbackHome.java b/Settings/src/com/android/tv/settings/system/FallbackHome.java
index 0924606..c45a88f 100644
--- a/Settings/src/com/android/tv/settings/system/FallbackHome.java
+++ b/Settings/src/com/android/tv/settings/system/FallbackHome.java
@@ -30,6 +30,10 @@ import android.util.Log;
 
 import java.util.Objects;
 
+import android.content.ContentResolver;
+import android.provider.Settings;
+import android.os.SystemProperties;
+
 public class FallbackHome extends Activity {
     private static final String TAG = "FallbackHome";
 
@@ -49,6 +53,11 @@ public class FallbackHome extends Activity {
     private BroadcastReceiver mReceiver = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
+            ContentResolver resolver = context.getContentResolver();
+            if (("1").equals(SystemProperties.get("ro.boot.firstboot"))) {
+                Settings.Secure.putInt(resolver, Settings.Secure.SHOW_IME_WITH_HARD_KEYBOARD, 1);
+            }
+            Settings.Secure.putInt(resolver, Settings.Secure.TV_USER_SETUP_COMPLETE, 1);
             maybeFinish();
         }
     };
-- 
1.9.1

